src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    const DAMAGE_OF_BITE = 50;
+    const isHerbivore = (animal instanceof Herbivore);
+    const isEnoughHealth = (animal.health >= DAMAGE_OF_BITE);
+    const isVisible = (animal.hidden === false);
+
+    if (isHerbivore && isVisible && isEnoughHealth) {
+      animal.health -= DAMAGE_OF_BITE;
+    }
```

Comment:
The animal should be bitten even if it has not enough health. It'll die anyway, right? ;(

Reply:
Yeah, my bad) 
I thought it wouldn't make sense for an animal with zero health to be bitten, but I agree that Carnivore can eat the rests of an animal. 
Thank you for the correction))

Reply:
But you were not checking if the animal has zero health. You were checking if the health is bigger than or equal to the bite damage. So if the animal has 20 health and the bite damage is 1000 the animal wouldn't die.

Reply:
Oh, indeed) I initially compared it to zero, and at the end, for some reason, I switched to comparing it with bite damage)  that's why I still thought I was comparing it to zero)